blog/2020-01-angular-scully.md

```diff
@@ -0,0 +1,332 @@
+
+<hr>
+
+**Table of contents:**
+
+- [About Scully](/blog/2020-01-angular-scully#about-scully)
+- [Get started](/blog/2020-01-angular-scully#get-started)
+- [Turn it into a blog](/blog/2020-01-angular-scully#turn-it-into-a-blog)
+- [Use the _Scully_ service](/blog/2020-01-angular-scully#use-the-scully-service)
+- [Fetch dynamic information from an API](/blog/2020-01-angular-scully#fetch-dynamic-information-from-an-api)
+- [Conclusion](/blog/2020-01-angular-scully#conclusion)
+
+> _Scully_ is currently just available within an early version.
+> This blog post is based on version _0.0.9_
+> However some of the commands or API calls used here may change in the future.
+> It’s my intention the keep the blog post as up-to-date as possible.
+
+## About Scully
+
+Scully is a static site generator (SSG) for Angular apps.
+Scully will go analyze a built Angular app and detect all the routes of the app.
+It will then call every route it found, visit the page in the browser, renders the page and finally put the static rendered page on your disk.
+This is what we are calling **pre-rendering**.
+The result is a structure ready for shipping at your web server.
+
+All pre-rendered pages contain just plain HTML and CSS.
+In fact when deploying it, a user will be able to instantly access all routes and see the content with almost no delay.
+The resulting sites are very small static sites in just a few KBs so that even the access from a mobile device with a very low bandwidth is insane fast.
+It's significantly faster compared to the hundreds of KBs that you are downloading when calling a “normal” Angular app.
+
+But that’s not all: Once the pre-rendered page is shipped to the user, _Scully_ loads and bootstraps the “real” Angular app  in the background on top of the existing view.
+In fact _Scully_ will unite two great things:
+The power of pre-rendering and very fast access to sites and the power of a fully functional SPA written in Angular.
+
+## Get started
+
+The first thing we have to do is to setup our Angular app.
+To use _Scully_ later, we have to use an Angular version _9.0.0-rc.0_ or greater.
+As _Scully_ detects the content from the routes, we need to configure the Angular router as well.
+Therefore we add the appropriate flag `--routing` (we can also choose this option when the CLI prompts us).
+
+```bash
+npx -p @angular/cli@^9.0.0-rc ng new scully-blog --routing
+cd scully-blog  # navigate into the project
+```
+
+The next step is to setup our static site generator _Scully_.
+Therefore we are using the provided Angular schematic:
+
+```bash
+ng add @scullyio/init  # add _Scully_ to the project
+```
+
+Et voilà here it is: We now have a very minimalistic Angular app that uses the power of _Scully_ to automatically find all app routes, visit them and generate static sites out of it.
+We can already try it out by building our site and run _Scully_ over it:
```

Comment:
`It's ready for us to preview. Let's try it out by building our site and running _Scully_.`